devito/types/sync.py

```diff
@@ -0,0 +1,328 @@
+"""
+Support types to implement asynchronous execution via e.g., threading,
```

Comment:
The comment appears to conflict with the file title. Add another sentence of explanation for us mere mortals?

Reply:
OK, improving docstring